de.metas.purchasecandidate.base/src/main/java/de/metas/purchasecandidate/BPPurchaseScheduleRepository.java

```diff
@@ -246,4 +270,96 @@ private AllBPPurchaseSchedule(@NonNull final BPartnerId bpartnerId, @NonNull fin
 					.findFirst();
 		}
 	}
+
+	public void changeLeadTimeOffset(final BPartnerId bpartnerId, int leadTimeOffset)
+	{
+		final BPPurchaseSchedule bpPurchaseSchedule = getByBPartnerIdAndValidFrom(bpartnerId, SystemTime.asLocalDate())
+														.orElse(null);
+		if (bpPurchaseSchedule == null)
+		{
+			throw new AdempiereException("BPPurchaseSchedule doess not exists for partner {}" + BPPurchaseSchedule.class + ": " + bpartnerId.toString());
+		}
+		else
+		{
+			bpPurchaseSchedule.toBuilder()
+			.leadTimeOffset(leadTimeOffset)
+			.build();
+
+			createOrUpdateAndSaveBPPurchaseScheduleRecord(bpPurchaseSchedule);
+		}
+	}
+
+
+	public I_C_BP_PurchaseSchedule createOrUpdateAndSaveBPPurchaseScheduleRecord(@NonNull final BPPurchaseSchedule schedule)
+	{
+		final I_C_BP_PurchaseSchedule scheduleRecord;
+		if (schedule.getBpPurchaseScheduleId() != null)
+		{
+			final int repoId = schedule.getBpPurchaseScheduleId().getRepoId();
+			scheduleRecord = load(repoId, I_C_BP_PurchaseSchedule.class);
+		}
+		else
+		{
+			scheduleRecord = newInstance(I_C_BP_PurchaseSchedule.class);
+		}
+
+		scheduleRecord.setC_BPartner_ID(BPartnerId.toRepoIdOr(schedule.getBpartnerId(),0));
+		scheduleRecord.setValidFrom(TimeUtil.asTimestamp(schedule.getValidFrom()));
+		scheduleRecord.setLeadTimeOffset(schedule.getLeadTimeOffset());
+		scheduleRecord.setReminderTimeInMin((int)schedule.getReminderTime().toMinutes());
+		final Frequency frequency = schedule.getFrequency();
+		scheduleRecord.setFrequencyType(toFrequencyTypeString(frequency.getType()));
+		if (frequency.getType() == FrequencyType.Weekly)
+		{
+			scheduleRecord.setFrequency(frequency.getEveryNthWeek());
+		}
+		else if (frequency.getType() == FrequencyType.Monthly)
+		{
+			scheduleRecord.setFrequency(frequency.getEveryNthMonth());
+			scheduleRecord.setMonthDay(frequency.getOnlyDaysOfMonth()
+					.stream()
+					.findFirst()
+					.orElseThrow(() -> new AdempiereException("No month of the day " + Frequency.class + ": " + frequency)));
+		}
+
+		setDaysOfWeek(scheduleRecord, frequency);
+
+		saveRecord(scheduleRecord);
+
+		return scheduleRecord;
+	}
+
+	private void setDaysOfWeek(@NonNull final I_C_BP_PurchaseSchedule scheduleRecord, final Frequency frequency)
```

Comment:
IMHO, instead of `Frequency frequency` param, i would have only `Set<DayOfWeek>`.

Reply:
i would also make it static